apm-agent-plugins/apm-awslambda-plugin/src/main/java/co/elastic/apm/agent/awslambda/helper/APIGatewayProxyV1TransactionHelper.java

```diff
@@ -86,27 +101,57 @@ private String getQueryString(APIGatewayProxyRequestEvent apiGatewayEvent) {
 
     @Override
     public void captureOutputForTransaction(Transaction transaction, APIGatewayProxyResponseEvent responseEvent) {
-        fillHttpResponseData(transaction, responseEvent.getHeaders(), responseEvent.getStatusCode());
+        Integer statusCode = responseEvent.getStatusCode();
+        if (statusCode == null) {
+            statusCode = 0;
+        }
+        fillHttpResponseData(transaction, responseEvent.getHeaders(), statusCode);
     }
 
     @Override
     protected void setTransactionTriggerData(Transaction transaction, APIGatewayProxyRequestEvent apiGatewayRequest) {
-        super.setTransactionTriggerData(transaction,apiGatewayRequest);
+        super.setTransactionTriggerData(transaction, apiGatewayRequest);
         APIGatewayProxyRequestEvent.ProxyRequestContext rContext = apiGatewayRequest.getRequestContext();
 
         if (null != rContext) {
-            setApiGatewayContextData(transaction, rContext.getRequestId(), rContext.getApiId(), rContext.getHttpMethod(),
-                    rContext.getResourcePath(), rContext.getAccountId());
+            setApiGatewayContextData(transaction, rContext.getRequestId(), rContext.getApiId(),
+                getHost(apiGatewayRequest), rContext.getAccountId());
         }
     }
 
     @Override
-    protected String getHttpMethod(APIGatewayProxyRequestEvent event) {
-        return event.getHttpMethod();
+    protected String getApiGatewayVersion() {
+        return "1.0";
     }
 
     @Override
-    protected String getApiGatewayVersion() {
-        return "1.0";
+    protected void setTransactionName(Transaction transaction, APIGatewayProxyRequestEvent event, Context lambdaContext) {
+        StringBuilder transactionName = transaction.getAndOverrideName(AbstractSpan.PRIO_HIGH_LEVEL_FRAMEWORK);
+        if (null != transactionName && requiredDataForTransactionNameAvailable(event)) {
+            transactionName.append(getHttpMethod(event)).append(" ");
+            if (webConfiguration.isUsePathAsName()) {
+                transactionName.append(event.getRequestContext().getPath());
+            } else {
+                setResourcePathBasedName(event, transactionName);
+            }
+        } else {
+            super.setTransactionName(transaction, event, lambdaContext);
+        }
```

Comment:
Some clarification questions:
1. if `webConfiguration.isUsePathAsName()` - shouldn't it be used anyway?
2. wouldn't a best-effort transaction naming approach be better than the all-or-nothing approach through `requiredDataForTransactionNameAvailable`?

Also, I see that you extracted this from `AbstractAPIGatewayTransactionHelper` into the concrete helpers, but there seem to still be a lot of shared logic.